src/py/flwr/common/address_test.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2020 Adap GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Flower IP address utils."""
+
+
+from .address import parse_address
+
+
+def test_ipv4_correct() -> None:
+    """Test if a correct IPv4 address is correctly parsed."""
+
+    # Prepare
+    addresses = [
```

Comment:
@tanertopal, are there any other cases like `localhost` that we should consider?

Reply:
Things we could test for:
```
("flower.dev:123", ("flower.dev", 123, False)),
("localhost:123", ("localhost", 123, False)),
("[::]:123", ("::", 123, True)),
("[0000:0000:0000:0000:0000:0000:0000:0001]:443", ("0000:0000:0000:0000:0000:0000:0000:0001", 443, True)),
("[0:0:0:0:0:0:0:1]:80", ("0:0:0:0:0:0:0:1", 80, True)),
("[::1]:80", ("::1", 80, True)),
("[0000:0000:0000:0000:0000:0000:0000:0001]:90", ("0000:0000:0000:0000:0000:0000:0000:0001", 90, True)),
```

The square brackets on the IPv6 are only needed when passing the port with the string. It's to make sure one can separate the port from the rest.

Reply:
Just saw the next comment. Probably some of these are for the other test.

Reply:
Also @charlesbvll the first two are not IPs but rather domains. They are still valid addresses which might resolve to either Ipv4 or Ipv6.

Reply:
@tanertopal Should I use something like `socket.gethostbyname(<DOMAIN_NAME>)` to get the actual IP before validating it? Otherwise, it doesn't really make sense to validate a domain. So, should I change the tests so that instead of returning the domain and the port, it returns the IP and the port?

Reply:
I'd propose that the parse function should return the host in it's original form (either and IP address or a host string like `flower.dev` or `localhost`). Wdyt @charlesbvll @tanertopal ?

Reply:
Yes, you're right, that makes more sense, I changed it already.